tools/pipeline-generator/Azure.Sdk.Tools.PipelineGenerator/Program.cs

```diff
@@ -236,5 +246,36 @@ private IEnumerable<SdkComponent> ScanForComponents(string path, string searchPa
             var components = scanner.Scan(scanDirectory, searchPattern);
             return components;
         }
+
+        private bool HasPipelineDefinitionNameDuplicates(PipelineConvention convention, IEnumerable<SdkComponent> components)
+        {
+            var pipelineNames = new Dictionary<string, SdkComponent>();
+            var duplicates = new HashSet<SdkComponent>();
+
+            foreach (var component in components)
+            {
+                var definitionName = convention.GetDefinitionName(component);
```

Comment:
IIRC this causes a network call each time and we already do this in other places. Should we just add the dictionary building and checking there instead?

Reply:
Somewhere in CreateOrUpdateDefinitionAsync?

Reply:
`GetDefinitionName` is just a string building function, for example: https://github.com/Azure/azure-sdk-tools/blob/baefa12425b2ba30e5f60447655f1a295c81af3e/tools/pipeline-generator/Azure.Sdk.Tools.PipelineGenerator/Conventions/PullRequestValidationPipelineConvention.cs#L19

It's less efficient to do a separate check here as opposed to adding this logic incrementally in the `scan` step, but I think it's simpler this way and won't have any noticeable impact.

Reply:
OK as long as we are just processing things in memory and not making more network calls I'm cool with making it an explicit check.

Reply:
I also explicitly did it here vs. per update call, because I wanted to be able to collect all collisions in case there are more than 2, so that we can fail with full information about what needs to be fixed.